From bb0c36a7b0aee5912d0a99e86d744dc096e5dd77 Mon Sep 17 00:00:00 2001
From: Nathan Chancellor <natechancellor@gmail.com>
Date: Mon, 8 Oct 2018 15:48:04 -0700
Subject: [PATCH 14/14] DO-NOT-UPSTREAM: samples: vfs: Disable for Clangi

Doesn't build right now.

In file included from samples/vfs/test-statx.c:28:
In file included from /usr/include/sys/stat.h:446:
/usr/include/bits/statx.h:25:8: error: redefinition of 'statx_timestamp'
struct statx_timestamp
       ^
./usr/include/linux/stat.h:56:8: note: previous definition is here
struct statx_timestamp {
       ^
In file included from samples/vfs/test-statx.c:28:
In file included from /usr/include/sys/stat.h:446:
/usr/include/bits/statx.h:36:8: error: redefinition of 'statx'
struct statx
       ^
./usr/include/linux/stat.h:99:8: note: previous definition is here
struct statx {
       ^
samples/vfs/test-statx.c:36:9: error: static declaration of 'statx'
follows non-static declaration
ssize_t statx(int dfd, const char *filename, unsigned flags,
        ^
/usr/include/bits/statx.h:87:5: note: previous declaration is here
int statx (int __dirfd, const char *__restrict __path, int __flags,
    ^
3 errors generated.
make[2]: *** [scripts/Makefile.host:90: samples/vfs/test-statx] Error 1
make[1]: *** [scripts/Makefile.build:520: samples/vfs] Error 2

Signed-off-by: Nathan Chancellor <natechancellor@gmail.com>
---
 samples/Kconfig | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/samples/Kconfig b/samples/Kconfig
index 8df1c012820f..0fde211b020b 100644
--- a/samples/Kconfig
+++ b/samples/Kconfig
@@ -148,7 +148,7 @@ config SAMPLE_VFIO_MDEV_MBOCHS
 
 config SAMPLE_VFS
 	bool "Build example programs that use new VFS system calls"
-	depends on X86
+	depends on X86 && !CC_IS_CLANG
 	help
 	  Build example userspace programs that use new VFS system calls such
 	  as mount API and statx().  Note that this is restricted to the x86
-- 
2.19.0

